examples/graph_sim/bench/graph_bench.cpp

```diff
@@ -0,0 +1,201 @@
+  edge.edge_info = std::string(len, in.id % 26 + 'b');
+  return edge;
+}
+
+static double NowSecs() {
+  struct timeval tv;
+  gettimeofday(&tv, nullptr);
+  return static_cast<double>(tv.tv_usec / 1000000.0) + tv.tv_sec;
+}
+
+// Simple timer to record some stats' time.
+class Timer {
+ public:
+  Timer(const std::string& name) : start_(NowSecs()), bench_name_(name) {}
+  ~Timer() { bench_timer.insert({bench_name_, NowSecs() - start_}); }
+
+ private:
+  double start_;
+  std::string bench_name_;
+};
+
+static void StartThreads(std::function<void()> func, int32_t thread_nums) {
+  std::vector<std::thread> ths;
+  for (int i = 0; i < thread_nums; i++) ths.emplace_back(func);
+
+  for (auto& t : ths) t.join();
+}
+
+void GraphDataConstruct() {
+  assert(FLAGS_client_threads != 0);
+  int64_t current_vertexes = FLAGS_vertex_nums / FLAGS_client_threads;
+
+  while (current_vertexes > 0) {
+    Status s;
+    uint64_t first_id, second_id;
+    int32_t vertex_info_len = FLAGS_vertex_info_len;
+    int32_t vertex_id_range = FLAGS_vertex_id_range;
+    first_id = generator();
+    second_id = generator();
+    if (vertex_id_range) {
+      first_id = first_id % vertex_id_range;
+      second_id = second_id % vertex_id_range;
+    }
+
+    Vertex src = CreateVertex(first_id, vertex_info_len);
+    Vertex dst = CreateVertex(second_id, vertex_info_len);
+
+    s = graph_simulator->AddVertex(src);
+    assert(s == Status::Ok);
+    s = graph_simulator->AddVertex(dst);
+    assert(s == Status::Ok);
+
+    Edge out_edge = CreateEdge(src, dst, 1, FLAGS_edge_info_len);
+    graph_simulator->AddEdge(out_edge);
+    Edge in_edge = CreateEdge(src, dst, 0, FLAGS_edge_info_len);
+    graph_simulator->AddEdge(in_edge);
+
+    current_vertexes -= 2;
+  }
+}
+
+void GraphDataSearchWithDegree() {
+  int32_t target_level = FLAGS_degree_level;
+  int32_t element_nums = FLAGS_degree_nums;
+
+  std::vector<Vertex> elements;
+
+  for (int i = 0; i < element_nums; i++) {
+    Vertex v;
+    auto s = graph_simulator->GetVertex(generator() % FLAGS_vertex_id_range, v);
+    // Not found, and search until find.
+    while (s != Status::Ok) {
+      s = graph_simulator->GetVertex(generator() % FLAGS_vertex_id_range, v);
+    }
+    elements.emplace_back(v);
+  }
+
+  std::vector<Status> status;
+  int32_t correct = 0;
+  graph_simulator->BFSSearch(elements, target_level, &status);
+  for (auto& state : status) {
+    if (state == Status::Ok) {
+      correct++;
+    }
+  }
+  fprintf(stdout, " success bfs search count : %d\n", correct);
+}
+
+void GraphDataTopN() {
+  assert(FLAGS_topn);
+  std::vector<std::pair<Vertex, uint64_t>> topn_res;
+  auto s = graph_simulator->GetTopN(topn_res, FLAGS_topn_num);
+  if (s != Status::Ok) {
+    fprintf(stdout, "Get the GraphData TopN Failed.");
+    return;
+  }
+
+  fprintf(stdout, "TopN: \n");
+  for (auto& item : topn_res) {
+    fprintf(stdout, "vertex : %llu  InEdges : %llu\n", item.first.id, item.second);
```

Comment:
Change %llu to %lu to avoid compile warning.

Reply:
agree,change to PRIu64